include/boost/math/interpolators/detail/cubic_hermite_detail.hpp

```diff
@@ -31,15 +33,21 @@ class cubic_hermite_detail {
         using std::isnan;
         if (x_.size() != y_.size())
         {
-            throw std::domain_error("There must be the same number of ordinates as abscissas.");
+            error_msg_ = "There must be the same number of ordinates as abscissas.";
+            valid_ = false;
+            return;
```

Comment:
So that we don't change the existing behavior what about something like:

```
boost::math::policies::raise_domain_error(...)
error_msg_ = ...;
valid_ = false;
return;
```

The raise domain error will throw when exceptions are enabled, and when exceptions are disabled we use track the validity of the state. This is the implementation of raise_domain_error: https://github.com/boostorg/math/blob/develop/include/boost/math/policies/error_handling.hpp#L208-L222. Like the comment says you shouldn't be able to hit the actual return value, but the continued implementations below will write to errno, NOP, etc. depending on the policy.

Reply:
+1 on that.